src/udev/udev-watch.c

```diff
@@ -20,57 +20,61 @@
 #include "udev-watch.h"
 
 int udev_watch_restore(int inotify_fd) {
-        DIR *dir;
+        _cleanup_closedir_ DIR *dir = NULL;
         int r;
 
         /* Move any old watches directory out of the way, and then restore the watches. */
 
         assert(inotify_fd >= 0);
 
         if (rename("/run/udev/watch", "/run/udev/watch.old") < 0) {
```

Comment:
hmm, for robustness reasons shouldn't this first try to `rm_rf("/run/udev/watch.old")` in case the dir still remains from a previous invocation of udev where we failed to remove the thing?